tests/test_packages/test_connections/test_p2p_libp2p/base.py

```diff
@@ -323,55 +332,52 @@ class BaseP2PLibp2pTest:
     """Base class for ACN p2p libp2p tests"""
 
     cwd: str
-    t: str
+    tmp: str
     tmp_dir: str
     tmp_client_dir: str
     log_files: List[str] = []
     multiplexers: List[Multiplexer] = []
     capture_log = True
 
+    libp2p_crypto = cosmos_crypto = make_crypto(CosmosCrypto.identifier)
+    default_crypto = ethereum_crypto = make_crypto(EthereumCrypto.identifier)
+    _cryptos = (None, cosmos_crypto, ethereum_crypto)
+
     @classmethod
     def setup_class(cls):
         """Set the test up"""
-
         atexit.register(cls.teardown_class)
-        cls.cwd, cls.t = os.getcwd(), TEMP_LIBP2P_TEST_DIR
-        if Path(cls.t).exists():
+        cls.cwd, cls.tmp = os.getcwd(), TEMP_LIBP2P_TEST_DIR
+        if Path(cls.tmp).exists():
             cls.remove_temp_test_dir()
-        cls.tmp_dir = os.path.join(cls.t, "temp_dir")
-        cls.tmp_client_dir = os.path.join(cls.t, "temp_client_dir")
-        Path(cls.tmp_dir).mkdir(parents=True, exist_ok=True)
-        os.chdir(cls.t)
+        Path(cls.tmp).mkdir()
+        os.chdir(cls.tmp)
 
     @classmethod
     def teardown_class(cls):
         """Tear down the test"""
-
         logging.debug(f"Cleaning up {cls.__name__}")
-        cls.disconnect()
+        cls._disconnect()
         cls.multiplexers.clear()
         cls.log_files.clear()
         os.chdir(cls.cwd)
-        if Path(cls.t).exists():
+        if Path(cls.tmp).exists():
             cls.remove_temp_test_dir()
         logging.debug(f"Teardown of {cls.__name__} completed")
 
     @classmethod
-    def disconnect(cls):
+    def _disconnect(cls):
         """Disconnect multiplexers and their connections"""
-
         for mux in cls.multiplexers:
-            for con in mux.connections:
-                con.disconnect()
             mux.disconnect()
+        wait_for_condition(lambda: cls.all_disconnected, timeout=TIMEOUT)
```

Comment:
should not be necessary. But this will error out just in case something does go wrong, it is an additional assurance that everything has been disconnected properly. I prefer this over leaving a comment to inform my future self or another dev.